Code/GraphMol/MMPA/MMPA.cpp

```diff
@@ -293,8 +294,81 @@ static void addResult(std::vector<std::pair<ROMOL_SPTR, ROMOL_SPTR> >&
       }
     }
   }
-  if (!resFound)
+  if (!resFound) {
+    //std::cerr << "**********************" << std::endl;
+    // From rfrag.py
+    // now change the labels on sidechains and core
+    // to get the new labels, cansmi the dot-disconnected side chains
+    // the first fragment in the side chains has attachment label 1, 2nd: 2, 3rd: 3
+    // then change the labels accordingly in the core
+    std::map<unsigned int, int> canonicalAtomMaps;
+    if( side_chains.get() ) {
+      RWMol _side_chain(*(side_chains.get())); // canonicalize the context or the core?
+      std::vector<int> oldMaps(_side_chain.getNumAtoms(), 0);
+      
+      // clear atom labels (they are used in canonicalization)
+      //  and move them to dummy storage
+      for (ROMol::AtomIterator at = _side_chain.beginAtoms(); at != _side_chain.endAtoms();
+           ++at) {
+        int label = 0;
+        if ((*at)->getPropIfPresent(common_properties::molAtomMapNumber, label) ) {
+          (*at)->clearProp(common_properties::molAtomMapNumber);
+          oldMaps[(*at)->getIdx()] = label;
+        }
+      }
+
+      MolOps::sanitizeMol(_side_chain);
```

Comment:
given that `sanitizeMol()` can thrown exceptions, it seems dangerous (and unecessary?) to have it here.